runtimes/pythonrt/runner/tests/values_test.py

```diff
@@ -0,0 +1,143 @@
+import pb.autokitteh.values.v1.values_pb2 as pb
+from values import wrap, unwrap
+from collections import namedtuple
+from datetime import datetime, timedelta, UTC
+import pytest
+import google.protobuf.timestamp_pb2 as timestamp_pb2
+import google.protobuf.duration_pb2 as duration_pb2
+
+
+def check(u, **kwargs):
+    v = pb.Value(**kwargs)
+    assert wrap(u) == v
+    assert unwrap(v) == u
+
+
+def intv(n):
+    return pb.Value(integer=pb.Integer(v=n))
+
+
+@pytest.mark.parametrize(
+    "u,v",
+    [
+        (None, pb.Value(nothing=pb.Nothing())),
+        (True, pb.Value(boolean=pb.Boolean(v=True))),
+        (False, pb.Value(boolean=pb.Boolean(v=False))),
+        (1, intv(1)),
+        (1.0, pb.Value(float=pb.Float(v=1.0))),
+        ("meow", pb.Value(string=pb.String(v="meow"))),
+        (
+            [1, 2, 3],
+            pb.Value(
+                list=pb.List(
+                    vs=[intv(1), intv(2), intv(3)],
+                ),
+            ),
+        ),
+        (
+            set([1, 2, 3]),
+            pb.Value(
+                set=pb.Set(vs=[intv(1), intv(2), intv(3)]),
+            ),
+        ),
+        (
+            {"a": 1, "b": 2},
+            pb.Value(
+                dict=pb.Dict(
+                    items=[
+                        pb.Dict.Item(
+                            k=pb.Value(string=pb.String(v="a")),
+                            v=intv(1),
+                        ),
+                        pb.Dict.Item(
+                            k=pb.Value(string=pb.String(v="b")),
+                            v=intv(2),
+                        ),
+                    ],
+                )
+            ),
+        ),
+        (
+            namedtuple("Point", ["y", "x"])(2, 1),
+            pb.Value(
+                struct=pb.Struct(
+                    ctor=pb.Value(string=pb.String(v="Point")),
+                    fields={
+                        "x": intv(1),
+                        "y": intv(2),
+                    },
+                )
+            ),
+        ),
+        (
+            timedelta(days=1),
+            pb.Value(duration=pb.Duration(v=duration_pb2.Duration(seconds=86400))),
+        ),
+        (
+            datetime.strptime("09/19/22 13:55:26", "%m/%d/%y %H:%M:%S").astimezone(
+                tz=UTC
+            ),
+            pb.Value(
+                time=pb.Time(
+                    v=timestamp_pb2.Timestamp(
+                        seconds=1663595726,
+                        nanos=0,
+                    )
+                )
+            ),
+        ),
+    ],
+)
+def test_wrap(u, v):
+    assert wrap(u) == v
+    assert unwrap(v) == u
+
+
+def test_special_wraps():
+    lv = pb.Value(
```

Comment:
Why can't these be part of the `test_wrap` cases?

Reply:
these are asymmetric